src/batch/src/executor/iceberg_scan.rs

```diff
@@ -295,3 +224,239 @@ impl BoxedExecutorBuilder for IcebergScanExecutorBuilder {
         }
     }
 }
+
+struct PositionDeleteFilter {
```

Comment:
We should also provide a method to remove the entry from the map because once we finish filtering a data file, we can release the corresponding entry.